src/panels/lovelace/editor/card-editor/hui-card-editor.ts

```diff
@@ -0,0 +1,264 @@
+// tslint:disable-next-line
+import { HuiYamlEditor } from "../../components/hui-yaml-editor";
+import { fireEvent } from "../../../../common/dom/fire_event";
+import { EntityConfig } from "../../entity-rows/types";
+
+declare global {
+  interface HASSDomEvents {
+    "entities-changed": {
+      entities: EntityConfig[];
+    };
+    "config-changed": {
+      config: LovelaceCardConfig;
+    };
+  }
+}
+
+@customElement("hui-card-editor")
+export class HuiCardEditor extends LitElement {
+  @property() public hass?: HomeAssistant;
+  @property() public value?;
+
+  @property() private _yaml?: string | undefined;
+  @property() private _configElement?: LovelaceCardEditor;
+  @property() private _configElType?: string;
+  @property() private _GUImode: boolean = true;
+  // Error: Configuration broken - do not save
+  @property() private _error?: string;
+  // Warning: GUI editor can't handle configuration - ok to save
+  @property() private _warning?: string;
+  @property() private _loading: boolean = false;
+
+  public get yaml(): string {
+    return this._yaml || "";
+  }
+  public set yaml(_yaml: string) {
+    this._yaml = _yaml;
+    this._updateConfigElement();
+    // TODO: Defer this to next redraw? How?
+    setTimeout(() => this._yamlEditor.codemirror.refresh(), 1);
+    fireEvent(this, "config-changed", { config: this.config! });
+  }
+
+  public get config(): LovelaceCardConfig | undefined {
+    try {
+      return yaml.safeLoad(this.yaml);
+    } catch (err) {
+      this._error = err;
+      return { type: "error", error: err.message };
+    }
+  }
+  public set config(config: LovelaceCardConfig | undefined) {
+    this.yaml = yaml.safeDump(config);
+  }
+
+  public get error(): boolean {
+    return this._error !== undefined;
+  }
+
+  private get _yamlEditor(): HuiYamlEditor {
+    return this.shadowRoot!.querySelector("hui-yaml-editor")!;
+  }
+
+  protected render(): TemplateResult {
+    return html`
+      <div style="width: 100%;">
+        <div class="gui-editor" ?hidden="${this._GUImode === false}">
+          ${this._loading
+            ? html`
+                <paper-spinner
+                  ?active="${this._loading}"
+                  alt="Loading"
+                  class="center margin-bot"
+                ></paper-spinner>
+              `
+            : this._warning || this._error
+            ? ""
+            : this._configElement}
+        </div>
+        <div class="yaml-editor" ?hidden="${this._GUImode === true}">
+          <hui-yaml-editor
+            .hass="${this.hass}"
+            .value="${this.yaml}"
+            @yaml-changed="${this._handleYAMLChanged}"
+          ></hui-yaml-editor>
+          <div class="error">
+            ${this._error}
+          </div>
+          <div class="warning">
+            ${this._warning}
+          </div>
+        </div>
+        <div class="buttons">
+          <mwc-button
+            @click="${() => (this._GUImode = !this._GUImode)}"
```

Comment:
We should not define inline functions. Instead make it a class method. 

Inline functions will be re-created each time render is called and will always cause an update, because the newly created object has a different identity than the old function.